strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/logging/LoggingManagementController.java

```diff
@@ -0,0 +1,229 @@
+import org.carlspring.strongbox.services.DirectoryListingServiceImpl;
+
+import javax.inject.Inject;
+import javax.servlet.http.HttpServletRequest;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Optional;
+import java.util.concurrent.ForkJoinPool;
+
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+import org.apache.commons.io.FilenameUtils;
+import org.apache.commons.io.input.Tailer;
+import org.apache.commons.lang.StringUtils;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.actuate.autoconfigure.logging.LogFileWebEndpointProperties;
+import org.springframework.boot.actuate.logging.LogFileWebEndpoint;
+import org.springframework.boot.logging.LogFile;
+import org.springframework.context.ApplicationContext;
+import org.springframework.core.io.FileSystemResource;
+import org.springframework.core.io.Resource;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.ModelMap;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestHeader;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.servlet.ModelAndView;
+import org.springframework.web.servlet.mvc.method.annotation.SseEmitter;
+import static org.carlspring.strongbox.controllers.logging.LoggingManagementController.ROOT_CONTEXT;
+import static org.springframework.http.HttpStatus.BAD_REQUEST;
+
+/**
+ * This controllers provides a simple wrapper over REST API for the LoggingManagementService.
+ *
+ * @author Martin Todorov
+ * @author Pablo Tirado
+ * @author Aditya Srinivasan
+ * @author Przemyslaw Fusik
+ */
+@Controller
+@Api(value = ROOT_CONTEXT)
+@RequestMapping(ROOT_CONTEXT)
+@PreAuthorize("hasAnyAuthority('ADMIN')")
+public class LoggingManagementController
+        extends BaseController
+{
+
+    public final static String ROOT_CONTEXT = "/api/logging";
+
+    @Value("${strongbox.sse.timeoutMillis:600000}")
+    private Long sseTimeoutMillis;
+
+    @Inject
+    private PropertiesBooter propertiesBooter;
+
+    @Inject
+    private ApplicationContext applicationContext;
+
+    @Inject
+    private LoggingManagementService loggingManagementService;
+
+    @Inject
+    private Optional<LogFileWebEndpointProperties> logFileWebEndpointProperties;
+
+    private DirectoryListingService directoryListingService;
+
+    public DirectoryListingService getDirectoryListingService()
```

Comment:
@fuss86 can you clarify why this needed?

Reply:
This has been copied from the `BrowseController` as we need the same feature for the log files.

Reply:
it will be good if we can understand why this was needed and do we need it now.

Reply:
I'm not sure what you're asking - are you talking about the actual implementation? 
It was refactored by you as part of SB-1164. We could probably move this logic into the `BaseController` but I would suggest we open up a follow-up task for this. Is this what you had in mind? :)

Reply:
:+1: For a follow-up task, as this one is really growing...

Reply:
👍

Reply:
(I don't like this implementation too - we will need a follow up task for this for sure)

Reply:
Okay then - can one of you guys write a follow-up task for this with how you guys envision it to be refactored?

Reply:
#1542

Reply:
Thanks, @sbespalov ! :)